src/restoreImpl.ts

```diff
@@ -31,11 +31,16 @@ async function restoreImpl(
         const cachePaths = utils.getInputAsArray(Inputs.Path, {
             required: true
         });
+        const enableCrossOsArchive = utils.getInputAsBool(
+            Inputs.EnableCrossOsArchive
+        );
 
         const cacheKey = await cache.restoreCache(
             cachePaths,
             primaryKey,
-            restoreKeys
+            restoreKeys,
+            {},
```

Comment:
Just curious, why do we have `{}` here?

Reply:
Because there is a download options which is not being passed. https://github.com/actions/toolkit/blob/main/packages/cache/src/options.ts#L24

Reply:
Got it. So I think optional parameters such as DownloadOptions need to stay at the end so we don't need to do this. You might have to change the parameter position of `enableCrossOsArchive` to come before this.

Reply:
That would need change in toolkit. Does this mean we don't want `enableCrossOsArchive` to be optional?

Reply:
This could possibly be a breaking change for projects using toolkit. Then that would be minor version release

Reply:
Depends on how are we calling it. So there are two types of parameters we can have, 
1. Mandatory parameters that have a default value so explicitly setting them is optional
2. Optional parameters that can be empty

If both are optional parameters that can be empty then this problem is unavoidable.

Reply:
Going ahead with current approach